tests/MongoDB.Driver.Tests/Specifications/Runner/BaseTestRunner.cs

```diff
@@ -0,0 +1,391 @@
+                {
+                    var client = DriverTestConfiguration.Client;
+                    var database = client.GetDatabase(DatabaseName);
+                    var collection = database.GetCollection<BsonDocument>(CollectionName).WithWriteConcern(WriteConcern.WMajority);
+                    collection.InsertMany(documents);
+                }
+            }
+        }
+
+        protected void RunTests(JsonDrivenTestCase testCase)
+        {
+            ValidateServerRequirements(testCase.Shared);
+            Run(testCase.Shared, testCase.Test);
+        }
+
+        protected void VerifyCollectionData(IEnumerable<BsonDocument> expectedDocuments)
+        {
+            var database = DriverTestConfiguration.Client.GetDatabase(DatabaseName);
+            var collection = database.GetCollection<BsonDocument>(CollectionName);
+            var actualDocuments = collection.Find("{}").ToList();
+            actualDocuments.Should().BeEquivalentTo(expectedDocuments);
+        }
+        #endregion
+
+
+        #region Private methods
+        private void AssertEvent(object actualEvent, BsonDocument expectedEvent)
+        {
+            if (expectedEvent.ElementCount != 1)
+            {
+                throw new FormatException("Expected event must be a document with a single element with a name the specifies the type of the event.");
+            }
+
+            var eventType = expectedEvent.GetElement(0).Name;
+            var eventAsserter = EventAsserterFactory.CreateAsserter(eventType);
+            eventAsserter.AssertAspects(actualEvent, expectedEvent[0].AsBsonDocument);
+        }
+
+        private ReadPreference ReadPreferenceFromBsonValue(BsonValue value)
+        {
+            if (value.BsonType == BsonType.String)
+            {
+                var mode = (ReadPreferenceMode)Enum.Parse(typeof(ReadPreferenceMode), value.AsString, ignoreCase: true);
+                return new ReadPreference(mode);
+            }
+
+            return ReadPreference.FromBsonDocument(value.AsBsonDocument);
+        }
+
+        private void Run(BsonDocument shared, BsonDocument test)
+        {
+            if (test.Contains(SkipReasonKey))
+            {
+                throw new SkipException(test[SkipReasonKey].AsString);
+            }
+
+            DatabaseNameKey = Ensure.IsNotNullOrEmpty(DatabaseNameKey, nameof(DatabaseNameKey));
+            CollectionNameKey = Ensure.IsNotNullOrEmpty(CollectionNameKey, nameof(CollectionNameKey));
+
+            JsonDrivenHelper.EnsureAllFieldsAreValid(shared, ExpectedSharedColumns);
+            JsonDrivenHelper.EnsureAllFieldsAreValid(test, ExpectedTestColumns);
+            CustomDataValidation(shared, test);
+
+            DatabaseName = shared[DatabaseNameKey].AsString;
+            CollectionName = shared[CollectionNameKey].AsString;
+
+            TestInitialize();
+            DropCollection();
+            CreateCollection();
+            InsertData(shared);
```

Comment:
It depends on the tests. The scenario can be like: I've inserted one record and I want to check that a collection contains only inserted record. 
However it's not a problem to create some flag which will be overwritten in the inheritors to control this behavior.
It can be for example create and delete collection only once for each json file. But I'm not sure that it's a really necessary logic. We can discuss it.

Reply:
I'm not taking the time now to decide if this base class is perfect or not.

The only thing that matters right now is that it is sufficient for the TransactionsConvenientApiTestRunner.